src/appleseed/foundation/meta/tests/test_iesparser.cpp

```diff
@@ -0,0 +1,313 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Artem Bishev, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.foundation headers.
+#include "foundation/utility/iesparser.h"
+#include "foundation/utility/iostreamop.h"
+#include "foundation/utility/test.h"
+
+// Standard headers.
+#include <iostream>
+#include <sstream>
+#include <fstream>
+
+// Boost headers.
+#include "boost/lexical_cast.hpp"
+
+using namespace foundation;
+
+TEST_SUITE(Foundation_Utility_Iesparser)
+{
+    TEST_CASE(ReadLine)
+    {
+        IESParser parser;
+
+        {
+            std::istringstream input_stream(" \n\t\t\r\n    \nline1\n  \tline2  ");
+            parser.ignore_empty_lines = true;
+            parser.reset(input_stream);
+            EXPECT_EQ(parser.line, "line1");
+            EXPECT_EQ(parser.line_counter, 4);
+            parser.read_trimmed_line(input_stream);
+            EXPECT_EQ(parser.line, "line2");
+            EXPECT_EQ(parser.line_counter, 5);
+        }
+
+        {
```

Comment:
I suggest to split this test case in two, one with `ignore_empty_lines` set to true, one with it set to false.